core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycle.scala

```diff
@@ -134,10 +134,12 @@ class BpmBillingProjectLifecycle(
             )
             _ <- billingRepository.setBillingProfileId(createProjectRequest.projectName, profileModel.getId)
           } yield CreationStatuses.CreatingLandingZone).recoverWith { case t: Throwable =>
+            logger.error("Billing project creation failed, cleaning up landing zone", t)
```

Comment:
Logging the original exception here because we are hitting a case where `cleanupLandingZone` itself throws an exception, and that's the one that ultimately is shown to the user/logged in `BillingProjectOrchestrator`. Is there a cleaner way to do this though?

This is the output with these changes (and database that is missing the USER_EMAIL column):

```
rawls [INFO] [19:28:25.689] [scala-execution-context-global-56] o.b.d.r.b.BillingProjectOrchestrator - Creating billing project record [name=RawlsBillingProjectName(CARTerraNov304)]
rawls [INFO] [19:28:26.853] [scala-execution-context-global-81] o.b.d.r.b.BillingProjectOrchestrator - Created billing project record, running post-creation steps [name=CARTerraNov304]
rawls [INFO] [19:28:26.857] [scala-execution-context-global-56] o.b.d.r.b.BillingProfileManagerDAOImpl - Creating billing profile [id=3d400f95-1a7d-452a-958c-0a2c508107f9]
rawls [INFO] [19:28:34.181] [scala-execution-context-global-56] o.b.d.r.b.BpmBillingProjectLifecycle - Creating BPM-backed billing project CARTerraNov304, created profile with ID 3d400f95-1a7d-452a-958c-0a2c508107f9.
rawls [INFO] [19:28:36.398] [scala-execution-context-global-56] o.b.d.r.b.BpmBillingProjectLifecycle - Initiated creation of landing zone 73de7316-ec49-452d-a304-3b5095041682 with jobId 055fd708-9c51-4939-910a-05add7370dea
rawls [ERROR] [19:28:37.269] [scala-execution-context-global-56] o.b.d.r.b.BpmBillingProjectLifecycle - Billing project creation failed, cleaning up landing zone
rawls java.sql.SQLSyntaxErrorException: Unknown column 'USER_EMAIL' in 'field list'
rawls 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
rawls 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994)
rawls 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
rawls 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
rawls 	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction.$anonfun$run$15(JdbcActionComponent.scala:520)
rawls 	at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement(JdbcBackend.scala:427)
rawls 	at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement$(JdbcBackend.scala:422)
rawls 	at slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:491)
rawls 	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:511)
rawls 	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction.run(JdbcActionComponent.scala:517)
rawls 	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28)
rawls 	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:25)
rawls 	at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)
rawls 	at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)
rawls 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
rawls 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
rawls 	at java.base/java.lang.Thread.run(Thread.java:833)
rawls [INFO] [19:28:37.595] [scala-execution-context-global-66] o.b.d.r.b.BpmBillingProjectLifecycle - Initiated deletion of landing zone 73de7316-ec49-452d-a304-3b5095041682 for BPM-backed billing project CARTerraNov304.
rawls [ERROR] [19:28:37.596] [scala-execution-context-global-66] o.b.d.r.b.BpmBillingProjectLifecycle - Billing project creation failed, cleaning up billing profil
rawls java.sql.SQLSyntaxErrorException: Unknown column 'USER_EMAIL' in 'field list'
rawls 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
rawls 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320)
rawls 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994)
rawls 	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
rawls 	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
rawls 	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction.$anonfun$run$15(JdbcActionComponent.scala:520)
rawls 	at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement(JdbcBackend.scala:427)
rawls 	at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement$(JdbcBackend.scala:422)
rawls 	at slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:491)
rawls 	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:511)
rawls 	at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction.run(JdbcActionComponent.scala:517)
rawls 	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28)
rawls 	at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:25)
rawls 	at slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276)
rawls 	at slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276)
rawls 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
rawls 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
rawls 	at java.base/java.lang.Thread.run(Thread.java:833)
rawls [INFO] [19:28:38.123] [scala-execution-context-global-66] o.b.d.r.b.BpmBillingProjectLifecycle - Deleting BPM-backed billing project CARTerraNov304, deleting billing profile record 3d400f95-1a7d-452a-958c-0a2c508107f9
rawls [ERROR] [19:28:39.428] [scala-execution-context-global-66] o.b.d.r.b.BillingProjectOrchestrator - Error in post-creation steps for billing project [name=CARTerraNov304]
rawls bio.terra.profile.client.ApiException: {"message":"Error deleting a profile in Sam: Cannot delete a resource with children. Delete the children first then try again.","statusCode":400,"causes":[]}
rawls 	at bio.terra.profile.client.ApiClient.invokeAPI(ApiClient.java:745)
rawls 	at bio.terra.profile.api.ProfileApi.deleteProfile(ProfileApi.java:167)
rawls 	at org.broadinstitute.dsde.rawls.billing.BillingProfileManagerDAOImpl.deleteBillingProfile(BillingProfileManagerDAO.scala:114)
rawls 	at org.broadinstitute.dsde.rawls.billing.BpmBillingProjectLifecycle.$anonfun$cleanupBillingProfile$3(BpmBillingProjectLifecycle.scala:190)
rawls 	at scala.runtime.java8.JFunction1$mcVI$sp.apply(JFunction1$mcVI$sp.scala:18)
rawls 	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
rawls 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
rawls 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
rawls 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
rawls 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
rawls 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
rawls 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

Reply:
If I'm forced to chose one over the other, I'm of the mind that it's more important to see why the server couldn't do do the thing I asked it to do rather than a why it couldnt clean up after itself. I'd log the error thrown by `cleanup` as some apprpriate level so it's flagged to us and rethrow the original error.
In this case it's debateable which error is more useful.

Reply:
One point against logging and rethrowing - it may result in an exception being logged multiple times which makes the logs more messy imo

Reply:
Scrap all the above - I think this is good!

Reply:
I actually think your point about which exception to show is correct-- we should just log the exception from cleaning up, and show the original exception to the user. With the commit I just pushed up, that is now the behavior. User sees:

![image](https://user-images.githubusercontent.com/484484/205092657-91d18c52-ba0c-4914-adec-1ea66edbe6d1.png)


Logs, which also show the failure to delete the billing profile.
```
rawls [INFO] [15:20:14.373] [scala-execution-context-global-56] o.b.d.r.b.BillingProjectOrchestrator - Created billing project record, running post-creation steps [name=CARTerraDec1a]
rawls [INFO] [15:20:14.377] [scala-execution-context-global-77] o.b.d.r.b.BillingProfileManagerDAOImpl - Creating billing profile [id=f7aa35dd-bc67-4b83-a364-cd9e2df90dcb]
rawls [INFO] [15:20:21.682] [scala-execution-context-global-77] o.b.d.r.b.BpmBillingProjectLifecycle - Creating BPM-backed billing project CARTerraDec1a, created profile with ID f7aa35dd-bc67-4b83-a364-cd9e2df90dcb.
rawls [INFO] [15:20:23.693] [scala-execution-context-global-77] o.b.d.r.b.BpmBillingProjectLifecycle - Initiated creation of landing zone a4f082e7-37e5-4c4f-b527-2b86f5a9f38c with jobId 06e08bba-6fab-4c32-b40a-72f21a3de1d1
rawls [ERROR] [15:20:24.534] [scala-execution-context-global-77] o.b.d.r.b.BpmBillingProjectLifecycle - Billing project creation failed, cleaning up landing zone
rawls [INFO] [15:20:24.887] [scala-execution-context-global-56] o.b.d.r.b.BpmBillingProjectLifecycle - Initiated deletion of landing zone a4f082e7-37e5-4c4f-b527-2b86f5a9f38c for BPM-backed billing project CARTerraDec1a.
rawls [ERROR] [15:20:24.889] [scala-execution-context-global-77] o.b.d.r.b.BpmBillingProjectLifecycle - Billing project creation failed, cleaning up billing profile
rawls [INFO] [15:20:25.387] [scala-execution-context-global-77] o.b.d.r.b.BpmBillingProjectLifecycle - Deleting BPM-backed billing project CARTerraDec1a, deleting billing profile record f7aa35dd-bc67-4b83-a364-cd9e2df90dcb
rawls [WARN] [15:20:26.657] [scala-execution-context-global-77] o.b.d.r.b.BpmBillingProjectLifecycle - Unable to delete billing profile zone with ID f7aa35dd-bc67-4b83-a364-cd9e2df90dcb for BPM-backed billing project CARTerraDec1a.
rawls bio.terra.profile.client.ApiException: {"message":"Error deleting a profile in Sam: Cannot delete a resource with children. Delete the children first then try again.","statusCode":400,"causes":[]}
rawls 	at bio.terra.profile.client.ApiClient.invokeAPI(ApiClient.java:745)
rawls 	at bio.terra.profile.api.ProfileApi.deleteProfile(ProfileApi.java:167)
rawls 	at org.broadinstitute.dsde.rawls.billing.BillingProfileManagerDAOImpl.deleteBillingProfile(BillingProfileManagerDAO.scala:114)
rawls 	at org.broadinstitute.dsde.rawls.billing.BpmBillingProjectLifecycle.$anonfun$cleanupBillingProfile$3(BpmBillingProjectLifecycle.scala:200)
rawls 	at scala.runtime.java8.JFunction1$mcVI$sp.apply(JFunction1$mcVI$sp.scala:18)
rawls 	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
rawls 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
rawls 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
rawls 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
rawls 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
rawls 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
rawls 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
rawls [ERROR] [15:20:26.664] [scala-execution-context-global-77] o.b.d.r.b.BillingProjectOrchestrator - Error in post-creation steps for billing project [name=CARTerraDec1a]
rawls org.broadinstitute.dsde.rawls.billing.BillingProjectCreationException: ErrorReport(rawls,Unknown column 'USER_EMAIL' in 'field list',Some(500 Internal Server Error),ArraySeq(),ArraySeq(com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120), com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122), com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916), com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061), com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009), com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320), com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994), com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61), com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java), slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction.$anonfun$run$15(JdbcActionComponent.scala:520), slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement(JdbcBackend.scala:427), slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement$(JdbcBackend.scala:422), slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:491), slick.jdbc.JdbcActionComponent$InsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:511), slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction.run(JdbcActionComponent.scala:517), slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:28), slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:25), slick.basic.BasicBackend$DatabaseDef$$anon$3.liftedTree1$1(BasicBackend.scala:276), slick.basic.BasicBackend$DatabaseDef$$anon$3.run(BasicBackend.scala:276), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635), java.base/java.lang.Thread.run(Thread.java:833)),Some(class java.sql.SQLSyntaxErrorException))
rawls 	at org.broadinstitute.dsde.rawls.billing.BpmBillingProjectLifecycle$$anonfun$$nestedInanonfun$postCreationSteps$5$1.$anonfun$applyOrElse$3(BpmBillingProjectLifecycle.scala:148)
rawls 	at cats.syntax.FlatMapOps$.$anonfun$$greater$greater$1(flatMap.scala:54)
rawls 	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:470)
rawls 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
rawls 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
rawls 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
rawls 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
rawls 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
rawls 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
rawls [ERROR] [15:20:26.686] [rawls-akka.actor.default-dispatcher-26] akka.actor.ActorSystemImpl - HttpMethod(POST) http://rawls-sbt:8080/api/billing/v2: 500 Internal Server Error entity: {"causes":[],"exceptionClass":"java.sql.SQLSyntaxErrorException","message":"Unknown column 'USER_EMAIL' in 'field list'","source":"rawls","stackTrace":[{"className":"com.mysql.cj.jdbc.exceptions.SQLError","fileName":"SQLError.java","lineNumber":120,"methodName":"createSQLException"},{"className":"com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping","fileName":"SQLExceptionsMapping.java","lineNumber":122,"methodName":"translateException"},{"className":"com.mysql.cj.jdbc.ClientPreparedStatement","fileName":"ClientPreparedStatement.java","lineNumber":916,"methodName":"executeInternal"},{"className":"com.mysql.cj.jdbc.ClientPreparedStatement","fileName":"ClientPreparedStatement.java","lineNumber":1061,"methodName":"executeUpdateInternal"},{"className":"com.mysql.cj.jdbc.ClientPreparedStatement","fileName":"ClientPreparedStatement.java","lineNumber":1009,"methodName":"executeUpdateInternal"},{"className":"com.mysql.cj.jdbc.ClientPreparedStatement","fileName":"ClientPreparedStatement.java","lineNumber":1320,"methodName":"executeLargeUpdate"},{"className":"com.mysql.cj.jdbc.ClientPreparedStatement","fileName":"ClientPreparedStatement.java","lineNumber":994,"methodName":"executeUpdate"},{"className":"com.zaxxer.hikari.pool.ProxyPreparedStatement","fileName":"ProxyPreparedStatement.java","lineNumber":61,"methodName":"executeUpdate"},{"className":"com.zaxxer.hikari.pool.HikariProxyPreparedStatement","fileName":"HikariProxyPreparedStatement.java","lineNumber":-1,"methodName":"executeUpdate"},{"className":"slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction","fileName":"JdbcActionComponent.scala","lineNumber":520,"methodName":"$anonfun$run$15"},{"className":"slick.jdbc.JdbcBackend$SessionDef","fileName":"JdbcBackend.scala","lineNumber":427,"methodName":"withPreparedStatement"},{"className":"slick.jdbc.JdbcBackend$SessionDef","fileName":"JdbcBackend.scala","lineNumber":422,"methodName":"withPreparedStatement$"},{"className":"slick.jdbc.JdbcBackend$BaseSession","fileName":"JdbcBackend.scala","lineNumber":491,"methodName":"withPreparedStatement"},{"className":"slick.jdbc.JdbcActionComponent$InsertActionComposerImpl","fileName":"JdbcActionComponent.scala","lineNumber":511,"methodName":"preparedInsert"},{"className":"slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$SingleInsertAction","fileName":"JdbcActionComponent.scala","lineNumber":517,"methodName":"run"},{"className":"slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction","fileName":"JdbcActionComponent.scala","lineNumber":28,"methodName":"run"},{"className":"slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction","fileName":"JdbcActionComponent.scala","lineNumber":25,"methodName":"run"},{"className":"slick.basic.BasicBackend$DatabaseDef$$anon$3","fileName":"BasicBackend.scala","lineNumber":276,"methodName":"liftedTree1$1"},{"className":"slick.basic.BasicBackend$DatabaseDef$$anon$3","fileName":"BasicBackend.scala","lineNumber":276,"methodName":"run"},{"className":"java.util.concurrent.ThreadPoolExecutor","fileName":"ThreadPoolExecutor.java","lineNumber":1136,"methodName":"runWorker"},{"className":"java.util.concurrent.ThreadPoolExecutor$Worker","fileName":"ThreadPoolExecutor.java","lineNumber":635,"methodName":"run"},{"className":"java.lang.Thread","fileName":"Thread.java","lineNumber":833,"methodName":"run"}],"statusCode":500}